lib/spack/spack/spec.py

```diff
@@ -2830,46 +2828,6 @@ def ensure_no_deprecated(root):
             msg += "    For each package listed, choose another spec\n"
             raise SpecDeprecatedError(msg)
 
-    def concretize(self, tests: Union[bool, Iterable[str]] = False) -> None:
```

Comment:
In the Monday discussion @tgamblin suggested to officially deprecate this function.

That means the circular import is there to stay in the v1.0.0 release.

We did not discuss how to remove it after v1.0.0... It would be really bad if that required a new major version release of Spack. Breaking API in `v1.1.0` feels wrong too.

I wish this PR wasn't blocked by discussions about what 1.0 really means.

Reply:
Yeah, but we can mentally subtract 1 from the numbers it outputs.